controllers/runnerpool_controller.go

```diff
@@ -172,6 +166,48 @@ func mergeMap(m1, m2 map[string]string) map[string]string {
 	return m
 }
 
+func (r *RunnerPoolReconciler) reconcileSecret(ctx context.Context, log logr.Logger, rp *meowsv1alpha1.RunnerPool) error {
+	s := &corev1.Secret{}
+	s.SetNamespace(rp.GetNamespace())
+	s.SetName(rp.GetRunnerSecretName())
+	op, err := ctrl.CreateOrUpdate(ctx, r.Client, s, func() error {
+		if expiresInStr, ok := s.Annotations[constants.RunnerSecretExpiresInAnnotationKey]; ok {
+			expiresIn, err := time.Parse(time.RFC3339, expiresInStr)
+			if err != nil {
+				log.Error(err, "failed to parse expires-in annotation")
+				return err
+			}
+			if expiresIn.After(time.Now().Add(5 * time.Minute)) {
```

Comment:
Why do you add five minutes here?

Reply:
I think that It needs to renew the token before it expires.
The five minutes was a random thought.

I realized that the following code might be better.
```go
expiresIn.Add(-5 * time.Minute).After(time.Now())
```